### ISSUE TITLE
[Feature request] (Multiple Request, info needed)

### ISSUE THREAD

**Issue by leoncrf** [FireCubeStudios/UnitedSets#16]
- Drag and paste between the tabs and show a new selected folder
- Remove background side borders that encompass open windows
- Keyboard shortcut
- Sessions like Microsoft Edge, so that it has separate programs in the taskbar and we can identify each session according to what is open and its respective icon (use an icon similar to the Edge sessions style, in which we have the main icon bigger and another small icon inside the larger icon to identify that it is a set of windows in UnitedSets).
In other words, it's nice to have several UnitedSets sessions, but without grouping them on the taskbar by hovering over them.b
- Window name on taskbar with name of current tab open in UnitedSets
- Ability to set different sizes in each open program, instead of just obeying the size of the UnitedSets window
- Button to lock program so that it does not close unexpectedly

**Comment by Get0457** [comment]
> Drag and paste between the tabs and show a new selected folder
* Drag between tabs is implemented and is available if you clone fro source. However, it is not yet in store release. I'm not sure if this is what you mean by "drag and paste."
* I'm not sure what you mean by "show a new selected folder"
> Remove background side borders that encompass open windows
I tried but it breaks some of the window. I might retry that again in the future. Will be tracked on #17 
- Keyboard shortcut
Unfortunately, it might be hard to impossible to implement shortcuts since you are focusing in other programs, not on United Sets. An idea is to implement as global hotkeys, but this will influence how other program works. For example, if we have `Ctrl + W` shortcut to close tab, if you use Microsoft Edge or other apps with tabs, it will close the tab on both United Sets and Microsoft Edge, which causes weird behavior. PR about this will be accepted if it will not have side effects on other software.
> Sessions like Microsoft Edge, so that it has separate programs in the taskbar and we can identify each session according to what is open and its respective icon (use an icon similar to the Edge sessions style, in which we have the main icon bigger and another small icon inside the larger icon to identify that it is a set of windows in UnitedSets).
In other words, it's nice to have several UnitedSets sessions, but without grouping them on the taskbar by hovering over them.b
Can you send us some examples? I'm not sure if [this](https://microsoftedge.microsoft.com/addons/detail/tab-session-manager/jkjjclfiflhpjangefhgfjhgfbhajadk) is what you mean.
> Window name on taskbar with name of current tab open in UnitedSets
Will be tracked on #18
> Ability to set different sizes in each open program, instead of just obeying the size of the UnitedSets window
I'm not sure how this will work out. If the program is bigger than United Sets window, what should we do? If the program is smaller than United Sets window, what should we do? It's not really possible to implement without knowing a clear behavior of what we should do.
> Button to lock program so that it does not close unexpectedly
I'm not sure what you mean. We can look into have an option to do that for close button on the tab. However, it is not possible to force the program not to quit itself if it wants to quit itself.

**Comment by leoncrf** [comment]
> Arraste e cole entre as guias e mostre uma nova pasta selecionada

About drag and paste, I mean that it could be possible to drag a file from one program to another, and at the same time visualize the program where we are dragging the file when we place the mouse over the tab of the target program, working just like the new windows file explorer tabs.

> Pode nos enviar alguns exemplos? Não sei se[é isso](https://microsoftedge.microsoft.com/addons/detail/tab-session-manager/jkjjclfiflhpjangefhgfjhgfbhajadk)que quer dizer.

[here]( https://www.google.com/search?q=sessions+edge+taskbar&tbm=isch&ved=2ahUKEwiMwef81qH7AhWWAbkGHTvOD-wQ2-cCegQIABAA&oq=sessions+edge+taskbar&gs_lcp=CgNpbWcQAzoECCMQJ1DSA1j5CmC4C2gAcAB4AIABpAGIAc0GkgEDNC40mAEAoAEBqgELZ3dzLXdpei1pbWfAAQE&sclient=img&ei=zuprY4yiEJaD5OUPu5y_4A4&bih=1037&biw=1666#imgrc=ZM_FAFc-VMmjOM)

Note that there are several Edges on the taskbar, each of them opens separately, they are not grouped on the taskbar. This can enable UnitedSets to use multiple separate sessions, and customize the taskbar icon for each session, so it's easier to see what's in each session UnitedSets. The custom icon can be the icon of the current program running in UnitedSets, that is, the icon appears automatically without requiring the user to customize the icon. Another idea for the icon, is that it works similarly to the Edge icon (a larger icon and another smaller icon inside the larger icon, the latter would be the UnitedSets icon, and the smaller one would be the icon of the current program open in UnitedSets) .

> I'm not sure how this will work out. If the program is bigger than United Sets window, what should we do? If the program is smaller than United Sets window, what should we do? It's not really possible to implement without knowing a clear behavior of what we should do.

Forget it, it's a bad idea I had.

> I'm not sure what you mean. We can look into have an option to do that for close button on the tab. However, it is not possible to force the program not to quit itself if it wants to quit itself.

Forget that idea too, I suggest a better one

-  When opening a UnitedSets session, having the possibility to have restore all the programs that were there previously

I forgot earlier to congratulate you on your work (:

**Comment by Get0457** [comment]
> About drag and paste, I mean that it could be possible to drag a file from one program to another, and at the same time visualize the program where we are dragging the file when we place the mouse over the tab of the target program, working just like the new windows file explorer tabs.

Yep, that is implemented but not yet on the store release

> [here]( https://www.google.com/search?q=sessions+edge+taskbar&tbm=isch&ved=2ahUKEwiMwef81qH7AhWWAbkGHTvOD-wQ2-cCegQIABAA&oq=sessions+edge+taskbar&gs_lcp=CgNpbWcQAzoECCMQJ1DSA1j5CmC4C2gAcAB4AIABpAGIAc0GkgEDNC40mAEAoAEBqgELZ3dzLXdpei1pbWfAAQE&sclient=img&ei=zuprY4yiEJaD5OUPu5y_4A4&bih=1037&biw=1666#imgrc=ZM_FAFc-VMmjOM)
> Note that there are several Edges on the taskbar, each of them opens separately, they are not grouped on the taskbar. This can enable UnitedSets to use multiple separate sessions, and customize the taskbar icon for each session, so it's easier to see what's in each session UnitedSets. The custom icon can be the icon of the current program running in UnitedSets, that is, the icon appears automatically without requiring the user to customize the icon. Another idea for the icon, is that it works similarly to the Edge icon (a larger icon and another smaller icon inside the larger icon, the latter would be the UnitedSets icon, and the smaller one would be the icon of the current program open in UnitedSets) .

It seems like different Edge icons are because they are actually different programs. One for normal edge, one for the beta edge program, and dev version, and one for canery version. I cannot really produce 4 executables for this.

We might look into creating different start menu icons instead. This might be possible. Will be tracked on #20

> When opening a UnitedSets session, having the possibility to have restore all the programs that were there previously

Will be tracked on #19 

I will close the issue since everything to be implemented will be in the separate issue.

**Comment by Get0457** [comment]
If you have any other suggestions or clarifications, feel free to open a new issue or talk about it in the specific issue I mentioned.
